WyattHWilliams/react-challenge-project-jan-2022
# #8 Add-Feature: edit & delete orders

## Changes
1. Added deleteOrder function to viewOrders component that makes an API call to delete. Passed that funciton in props to ordersList.
2. added edit-order/:id route for editing specific orders using the already-made order-form component
3. refactored the functionality in the useEffect call into its own function for readability and cleaner organization
4. changed the delete button in ordersList to use the deleteOrder function.
5. changed the edit button in ordersList to be a link to the edit/order route that passeds the order id in the url and the order object as state to the order-form component at the edit-order route location.
6. Modified the order-form component to include functionality for editing the order, and for checking to see if a new order is being made or if an old order is being changed

## Purpose
We wanted to add delete and edit functionality without creating any new components and without adding a whole bunch of code to keep it as "dry" as possible.
Addresses [issue 8](https://github.com/Shift3/react-challenge-project-jan-2022/issues/8)

## Approach
1. By reusing a slightly modified order-form component we avoid making any new components related to editing an order.
2. By using an edit-order/:id path we separate concerns
3. By using a link to the edit-order path we can pass the order as state directly to the order-form component without that component needing to be a child of OrdersList, and we get to persist the data for that order without adding to, or creating, a reducer.

## Pre-Testing TODOs
1. login the user

## Testing Steps
2. create an order
3. delete that order, check that the order is no longer rendered
4. refresh the view-orders page or make a get-all orders call to check that the deleted order actually got deleted from the server
5. create an order
6. click the edit order button in the view-orders page, check that user has arrived on edit-orders page
7. check that order id is in URL
8. check that the order information has been preloaded into the form
9. check that the form header has changed to "actually i'd like a..."
10. edit the order, click "order-it!"
11. check that user has been redirected to the view-orders page
12. check that order exists and has changed

Closes #008
